### DISCUSSION TITLE
Permissions for the token

### DISCUSSION THREAD

**Discussion by davidvalaz** [ArtiomTr/jest-coverage-report-action#58] [Q&A] [answered]
I would love to incorporate this into my organisations repositories as it ticks all of the boxes. However, what are the permissions that need to be granted for the token and what is the exact process for generating them?

**Comment by ArtiomTr** [comment, marked as answer]
Hello!

Each repository automatically has GITHUB_TOKEN, so in my opinion, it should suit for most of the cases. ([here is the docs](https://docs.github.com/en/actions/reference/authentication-in-a-workflow))

However, if you want to provide custom token, you can find instructions how to create it [here](https://docs.github.com/en/github/authenticating-to-github/creating-a-personal-access-token). For permissions, you can provide all the default GITHUB_TOKEN permissions ([documented here](https://docs.github.com/en/actions/reference/authentication-in-a-workflow#permissions-for-the-github_token)). Of course, this action do not uses all of them, but I have not tested yet which are required.

I think these permissions should be enough, but I'm not sure:

| No | Permission    | Access type |
|----|---------------|-------------|
|  1 | contents      | read        |
|  2 | checks        | read/write  |
|  3 | pull requests | read/write  |
| 4  | metadata      | read        |

I hope my explanation will help you!
